complex.c

```diff
@@ -1851,8 +1851,11 @@ nucomp_to_r(VALUE self)
     get_dat1(self);
 
     if (!k_exact_zero_p(dat->imag)) {
-        rb_raise(rb_eRangeError, "can't convert %"PRIsVALUE" into Rational",
-                 self);
+        VALUE imag = rb_check_convert_type_with_id(dat->imag, T_RATIONAL, "Rational", idTo_r);
```

Comment:
why convert `imag` to rational here just to pass it into `k_exact_zero_p()`? Wouldn't it be easier to just change the check `k_exact_zero_p(dat->imag)` to `f_zero_p(dat->imag)` instead?

Reply:
@KJTsanaktsidis This approach doesn’t impose the responsibility of deciding what exactly constitutes zero on the Complex class. Instead, it assigns the responsibility to each individual class.

Reply:
And this implementation is for proof of concept. Before merging this to the master branch, I’ll add the optimization for Float class later.

Reply:
Ah that makes sense - thanks for the explanation!